File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        a_count = 0
        b_count = 0
        clength = len(colors) - 1
        
        for i in range(1, clength):
            if colors[i-1] == 'A' and colors[i] == 'A' and colors[i+1] == 'A':
                a_count += 1
                
            if colors[i-1] == 'B' and colors[i] == 'B' and colors[i+1] == 'B':
                b_count += 1
        
        if a_count > b_count:
            return True
        
        return False
        
                
            

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.js
/**
 * @param {string} colors
 * @return {boolean}
 */
var winnerOfGame = function(colors) {
    let alice = 0;
    let bob = 0;
    const clength = colors.length - 1;
    
    for (let i = 1; i < clength; i++){
        if (colors[i-1] === colors[i] && colors[i] === colors[i+1]){
            if (colors[i] === 'A'){
                alice++;
            }
            else {
                bob++;
            }
        }
    }
    
    return alice > bob;
};